libraries/tock-register-interface/src/peripheral.rs

```diff
@@ -0,0 +1,278 @@
+/// }
+/// ```
+/// and an implementation of the trait for use on real systems (as opposed to
+/// unit test environments):
+/// ```ignore
+/// #[derive(Clone, Copy)]
+/// struct RealFoo<P: tock_registers::MmioPointer> { ... }
+/// impl<P: tock_registers::MmioPointer> Foo for RealFoo<P> { ... }
+/// ```
+///
+/// # Registers with bitfields (LongNames)
+/// Registers can have LongNames, which specify the values of bitfields they
+/// contain:
+/// ```ignore
+/// tock_registers::register_bitfields![u32,
+///     Control [ ... ]
+///     Pin [ ... ]
+///     Status [ ... ]
+///     Trigger [ ... ]
+/// ];
+/// tock_registers::peripheral! {
+///     Foo {
+///         // A bitfield is specified by putting the bitfield type in parenthesis
+///         // after the register's data type:
+///         0x0 => control: u32(Control::Register) { Write },
+///
+///         // Array registers can have bitfields as well. The bitfield applies to
+///         // each register in the array.
+///         0x4 => pins: [u8; 4](Pin::Register) { Read, Write },
+///
+///         // Registers can have different bitfields for read operations than for
+///         // write operations. In that case, specify the register's LongName on
+///         // the operation rather than the
+///         0x8 => aliased: u32 { Read(Status::Register), Write(Trigger::Register) },
+///     }
+/// }
+/// ```
+/// Specifying bitfields changes the LongNames in the generated trait:
+/// ```ignore
+/// trait Foo {
+///     type control<'s> = tock_registers::Register<DataType = u32>
+///         + tock_registers::Write<LongName = Control::Register>
+///     where
+///         Self: 's;
+///     fn control(&self) -> Self::control<'_>;
+///
+///     type pins<'s> = tock_registers::Register<DataType = [u8; 4]>
+///         + tock_registers::Read<LongName = Pins::Register>
+///         + tock_registers::Write<LongName = Pins::Register>
+///     where
+///         Self: 's;
+///     fn pins(&self) -> Self::pins<'_>;
+///
+///     type aliased<'s> = tock_registers::Register<DataType = u32>
+///         + tock_registers::Read<LongName = Status::Register>
+///         + tock_registers::Write<LongName = Trigger::Register>
+///     where
+///         Self: 's;
+///     fn aliased(&self) -> Self::aliased<'_>;
+/// }
+/// ```
+///
+/// # Padding
+/// If there is a gap between registers in a peripheral, you must declare that
+/// gap as a padding field:
+/// ```
+/// tock_registers::peripheral! {
+///     Foo {
+///         0x0 => register_a: u8 {},
+///
+///         // The padding field; specified by writing _ in place of the name
+///         // and omitting the data type and operation list.
+///         0x1 => _,
+///
+///         // The offset of the next field determines the size of the padding
+///         // (in this case, the padding is two bytes).
+///         0x3 => register_b: u8 {},
+///     }
+/// }
+/// ```
+/// Padding fields are ommitted from the generated trait.
+///
+/// # Doc comments
```

Comment:
Consider either:

- Removing this section and place the rustdoc comments inside the starting example (as it should be immediately apparent and obvious to users), or
- Combining with the `cfg` attributes section below, as doc comment support is an extension of supporting attributes in general.

Reply:
I did a bit of both. Added doc comments inside the starting example, then combined this with the `cfg` attribute section. I used the new combined section to document *where* the doc comments end up.